optuna/distributions.py

```diff
@@ -450,6 +453,12 @@ def _asdict(self) -> Dict:
         return d
 
 
+def _nanequal(value1: Any, value2: Any) -> bool:
+    value1_is_nan = isinstance(value1, Real) and np.isnan(float(value1))
+    value2_is_nan = isinstance(value2, Real) and np.isnan(float(value2))
```

Comment:
Can we write these lines like below?

```suggestion
    value1_is_nan = isinstance(value1, float) and np.isnan(value1)
    value2_is_nan = isinstance(value2, float) and np.isnan(value2)
```

Reply:
I'm not sure if we should support `Real` types other than `float`. This PR can handle other NaNs like `np.float32("nan")`, but I think it is very rare to suggest such values. What do you think?

Reply:
Thanks! Such cases (e.g. `np.float32("nan")`) completely slipped my mind. Let me resolve this comment.